network/tcp.go

```diff
@@ -0,0 +1,411 @@
+package network
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/dedis/cothority/log"
+)
+
+// NewTCPRouter returns a new Router using TCPHost as the underlying Host.
+func NewTCPRouter(sid *ServerIdentity) (*Router, error) {
+	h, err := NewTCPHost(sid.Address)
+	if err != nil {
+		return nil, err
+	}
+	r := NewRouter(sid, h)
+	return r, nil
+}
+
+// TCPConn implements the Conn interface using plain, unencrypted TCP.
+type TCPConn struct {
+	// The name of the endpoint we are connected to.
+	endpoint Address
+
+	// The connection used
+	conn net.Conn
```

Comment:
They return the interface type because they have to (`net.Listen` is a general function that can create all sorts of connections); here, the opposite is the case: We explicitly want a TCPConn (and that's what `net.Listen("tcp",address)` implicitly returns, too).